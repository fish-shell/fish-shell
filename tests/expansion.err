Array index out of bounds
fish: show "$foo[2]"
                 ^
Array index out of bounds
fish: show $foo[2]
                ^
Array index out of bounds
fish: show "$foo[1 2]"
                   ^
Array index out of bounds
fish: show $foo[1 2]
                  ^
Array index out of bounds
fish: show "$foo[2 1]"
                 ^
Array index out of bounds
fish: show $foo[2 1]
                ^
Invalid index value
fish: echo "$foo[d]"
                 ^
Invalid index value
fish: echo $foo[d]
                ^
Array index out of bounds
fish: echo ()[1]
              ^
Invalid index value
fish: echo ()[d]
              ^
