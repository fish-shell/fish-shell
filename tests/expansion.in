# Test expansion of variables

function show --description 'Prints argument count followed by arguments'
    echo (count $argv) $argv
end

set -l foo
show "$foo"
show $foo
show "prefix$foo"
show prefix$foo

show "$$foo"
show $$foo
show "prefix$$foo"
show prefix$$foo

set -l foo ''
show "$foo"
show $foo
show "prefix$foo"
show prefix$foo

show "$$foo"
show $$foo
show "prefix$$foo"
show prefix$$foo

set -l foo bar
set -l bar
show "$$foo"
show $$foo
show "prefix$$foo"
show prefix$$foo

set -l bar baz
show "$$foo"
show $$foo
show "prefix$$foo"
show prefix$$foo

set -l bar baz quux
show "$$foo"
show $$foo
show "prefix$$foo"
show prefix$$foo

set -l foo bar fooer fooest
set -l fooer
set -l fooest
show "$$foo"
show $$foo
show "prefix$$foo"
show prefix$$foo

set -l fooer ''
show $$foo
show prefix$$foo

set -l foo bar '' fooest
show "$$foo"
show $$foo
show "prefix$$foo"
show prefix$$foo

set -l foo
show "$foo[1]"
show $foo[1]
show "$foo[-1]"
show $foo[-1]
show "$foo[2]"
show $foo[2]
show "$foo[1 2]"
show $foo[1 2]
show "$foo[2 1]"
show $foo[2 1]

echo "$foo[d]"
echo $foo[d]

echo ()[1]
echo ()[d]
