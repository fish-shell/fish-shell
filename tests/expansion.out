1 
0
1 prefix
0
1 
0
1 prefix
0
1 
1 
1 prefix
1 prefix
1 
0
1 prefix
0
1 
0
1 prefix
0
1 baz
1 baz
1 prefixbaz
1 prefixbaz
1 baz quux
2 baz quux
1 prefixbaz quux
2 prefixbaz prefixquux
1 baz quux fooer fooest
2 baz quux
1 prefixbaz quux fooer fooest
2 prefixbaz prefixquux
3 baz quux 
3 prefixbaz prefixquux prefix
1 baz quux  fooest
2 baz quux
1 prefixbaz quux  fooest
2 prefixbaz prefixquux
1 
0
1 
0
